netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -88,6 +93,7 @@
     this.authority = checkNotNull(authority, "authority");
     this.scheme = checkNotNull(scheme, "scheme");
     this.userAgent = userAgent;
+    this.attrsProvider = attrsProvider;
```

Comment:
Instead of attributes provider (which may be null, which is strange), could we just call `state.handler.getAttributes()` (probably by adding a getAttributes() method on the TransportState and then calling state.getAttributes())? There'd be no reason to have `getAttrs()` on the transport either, then.